File: cakk-api/src/test/java/com/cakk/api/service/like/LikeServiceTest.java
package com.cakk.api.service.like;

import static org.junit.jupiter.api.Assertions.*;
import static org.mockito.Mockito.*;

import org.junit.jupiter.api.DisplayName;
import org.mockito.InjectMocks;
import org.mockito.Mock;

import net.jqwik.api.Arbitraries;

import com.cakk.api.common.annotation.TestWithDisplayName;
import com.cakk.api.common.base.ServiceTest;
import com.cakk.core.facade.cake.CakeShopReadFacade;
import com.cakk.core.facade.user.UserLikeFacade;
import com.cakk.domain.mysql.entity.shop.CakeShop;
import com.cakk.domain.mysql.entity.user.User;
@DisplayName("좋아요 기능 관련 비즈니스 로직 테스트")
public class LikeServiceTest extends ServiceTest {

	@InjectMocks
	private LikeService likeService;

	@Mock
	private CakeShopReadFacade cakeShopReadFacade;

	@Mock
	private UserLikeFacade userLikeFacade;

	@TestWithDisplayName("케이크 샵 좋아요를 성공한다.")
	void likeCakeShop() {
		// given
		final User user = getUser();
		final long cakeShopId = Arbitraries.longs().greaterOrEqual(1).sample();
		final CakeShop cakeShop = getConstructorMonkey().giveMeBuilder(CakeShop.class)
			.set("shopName", Arbitraries.strings().withCharRange('a', 'z').ofMaxLength(30))
			.set("shopBio", Arbitraries.strings().withCharRange('a', 'z').ofMaxLength(40))
			.set("shopDescription", Arbitraries.strings().withCharRange('a', 'z').ofMaxLength(500))
			.set("location", supplyPointBy(Arbitraries.doubles().sample(), Arbitraries.doubles().sample()))
			.sample();

		doReturn(cakeShop).when(cakeShopReadFacade).findByIdWithLike(cakeShopId);
		doNothing().when(userLikeFacade).likeCakeShop(user, cakeShop);

		// when
		likeService.likeCakeShop(user, cakeShopId);

		// then
		assertDoesNotThrow(() -> likeService.likeCakeShop(user, cakeShopId));
	}
}

File: cakk-core/src/test/kotlin/com/cakk/core/service/like/LikeServiceTest.kt
package com.cakk.core.service.like

import io.kotest.assertions.throwables.shouldNotThrow

import org.junit.jupiter.api.DisplayName
import org.mockito.InjectMocks
import org.mockito.Mock
import org.mockito.Mockito.doNothing
import org.mockito.Mockito.doReturn

import com.cakk.core.common.annotation.TestWithDisplayName
import com.cakk.core.common.base.MockitoTest
import com.cakk.core.common.fixture.FixtureCommon.fixtureMonkey
import com.cakk.core.common.fixture.FixtureCommon.getLongFixtureGoe
import com.cakk.core.common.fixture.FixtureCommon.getPointFixture
import com.cakk.core.common.fixture.FixtureCommon.getStringFixtureBw
import com.cakk.core.facade.cake.CakeShopReadFacade
import com.cakk.core.facade.user.UserLikeFacade
import com.cakk.domain.mysql.entity.shop.CakeShop

@DisplayName("좋아요 기능 관련 비즈니스 로직 테스트")
internal class LikeServiceTest : MockitoTest() {

	@InjectMocks
	private lateinit var likeService: LikeService

	@Mock
	private lateinit var cakeShopReadFacade: CakeShopReadFacade

	@Mock
	private lateinit var userLikeFacade: UserLikeFacade

	@TestWithDisplayName("케이크 샵 좋아요를 성공한다.")
	fun likeCakeShop() {
		// given
		val user = getUserFixture()
		val cakeShopId: Long = getLongFixtureGoe(1).sample()
		val cakeShop = fixtureMonkey.giveMeBuilder(CakeShop::class.java)
			.set("shopName", getStringFixtureBw(10, 30))
			.set("shopBio", getStringFixtureBw(10, 40))
			.set("shopDescription", getStringFixtureBw(100, 500))
			.set("location", getPointFixture())
			.sample()

		doReturn(cakeShop).`when`(cakeShopReadFacade).findByIdWithLike(cakeShopId)
		doNothing().`when`(userLikeFacade).likeCakeShop(user, cakeShop)

		// when
		likeService.likeCakeShop(user, cakeShopId)

		// then
		shouldNotThrow<Exception> { likeService.likeCakeShop(user, cakeShopId) }
	}
}
